javalib-ext-dummies/src/main/scala/java/security/SecureRandom.scala

```diff
@@ -0,0 +1,7 @@
+package java.security
+
+/** Fake implementation of `SecureRandom` that is not actually secure at all.
+ *
+ *  It directly delegates to `java.util.Random`.
+ */
+class SecureRandom extends java.util.Random
```

Comment:
@LeeTibbert I assume you are referring to this? This is not published. It is an internal dummy used for testing only.

Reply:
Yes, exactly.  I recommend posting 4 Marine's on duty and having a platoon as relief.

What I would do if I were telling the story, and I am not, is not even go near `secureRandom()` unless
it is indeed believably & defensibly secure, not even in testing or dummy code.  
You are handling smallpox virus outside a containment facility.

That said, I suggest  doing at least three things:

1)   Most importantly, I would have the file say, in better works, "For testing only"

2)   I would move the code to the unit-test itself and make it private to that class to make it even
      more evident  that it is for testing purposes and harder to take the blade guards off.

      The name `javalib-ext-dummies` requires Institutional Knowledge that even I do not have,
      not that I am all knowing and I have missed a lot in the past year. If were using 'find` to find
      a `SecureRandom()` implementation for a project and did not know the security implications
      you imply but do not describe for `java.util.Random` I would grab the `javalib-ext-dummies` file
      in a flash and complain loudly when it was not Secure.  I am not trying to be harsh here or
      to reveal my tendency to leap before I look.  

3) I would probably have `SecureRandom` directly call 
`MASSIVELY_INSECURE_RANDOM_USE_AT_YOUR_OWN_RISK_AND_DO_NOT_COMPLAIN_WHEN_NOT_IF" + "_YOU_HURT_YOURSELF()"

Reply:
I can definitely improve the comment. I don't think it should be moved to the unit test, that would not follow the existing pattern of `javalib-ext-dummies` (which of course can be challenged in a separate issue).

Note that this same pattern is used in https://github.com/scala-js/scala-js/pull/4659.

Reply:
I appreciate your considering my concerns and addressing them as is consistent with your
intent.

It is always an open issue as to when to follow precedent and when to take another path.  
The Scala.js folks have already said that they would not provide a  known insecure variant if they
had to do things a second time.  You have heard and considered my thoughts about `javalib-ext-dummies`
so I will not belabor them.

Reply:
> The Scala.js folks have already said that they would not provide a known insecure variant if they
had to do things a second time.

To clarify, they were talking about _publishing_ a known insecure variant.
https://github.com/scala-js/scala-js-fake-insecure-java-securerandom/

That is different from the internal dummy used for testing.

Reply:
re: `scala-js-fake-insecure-java-securerandom`  I like the name!   

Thank you for the chuckle, I could use one or a few.

Reply:
> > The Scala.js folks have already said that they would not provide a known insecure variant if they
> > had to do things a second time.
> 
> To clarify, they were talking about _publishing_ a known insecure variant. https://github.com/scala-js/scala-js-fake-insecure-java-securerandom/
> 
> That is different from the internal dummy used for testing.

Indeed. And to be even clearer, we would definitely do redo the same thing for testing. This is the best pattern to test "partial" implementations (`randomUUID`), which require some externally provided supplementary class (`SecureRandom`) to actually link.